components/dashboard/src/settings/Teams.tsx

```diff
@@ -0,0 +1,838 @@
+    const [pendingSlotsPurchase, setPendingSlotsPurchase] = useState<{ ts: TeamSubscription } | undefined>(getLocalStorageObject<{ ts: TeamSubscription }>("pendingSlotsPurchase"));
+
+    useEffect(() => {
+        queryState();
+    }, []);
+
+    useEffect(() => {
+        if (pendingPlanPurchase) {
+            setLocalStorageObject("pendingPlanPurchase", pendingPlanPurchase);
+        } else {
+            removeLocalStorageObject("pendingPlanPurchase");
+        }
+    }, [pendingPlanPurchase]);
+
+    useEffect(() => {
+        if (pendingSlotsPurchase) {
+            setLocalStorageObject("pendingSlotsPurchase", pendingSlotsPurchase);
+        } else {
+            removeLocalStorageObject("pendingSlotsPurchase");
+        }
+    }, [pendingSlotsPurchase]);
+
+    const queryState = async () => {
+        const [slots, showPaymentUI, teamSubscriptions, clientRegion, isStudent] = await Promise.all([
+            getGitpodService().server.tsGetSlots(),
+            getGitpodService().server.getShowPaymentUI(),
+            getGitpodService().server.tsGet(),
+            getGitpodService().server.getClientRegion(),
+            getGitpodService().server.isStudent(),
+        ]);
+
+        setDefaultCurrency((clientRegion && (countries as any)[clientRegion]?.currency === 'EUR') ? 'EUR' : 'USD');
+        setIsStudent(isStudent);
+
+        setSlots(slots);
+        setShowPaymentUI(showPaymentUI);
+        setTeamSubscriptions(teamSubscriptions);
+    }
+
+    useEffect(() => {
+        if (!isChargebeeCustomer) {
+            (async () => {
+                setIsChargebeeCustomer(await getGitpodService().server.isChargebeeCustomer());
+            })()
+        }
+        if (pendingPlanPurchase) {
+            if (teamSubscriptions.some(ts => ts.planId === pendingPlanPurchase.plan.chargebeeId)) {
+                setPendingPlanPurchase(undefined);
+            }
+        }
+    }, [teamSubscriptions]);
+
+    useEffect(() => {
+        if (pendingSlotsPurchase) {   
+            if (slots.some(s => s.teamSubscription.id === pendingSlotsPurchase.ts.id)) {
+                setPendingSlotsPurchase(undefined);
+            }
+        }
+    }, [slots]);
+
+    const getSubscriptionTypes = () => isStudent ? ['professional', 'professional-new', 'student'] : ['professional', 'professional-new'];
+
+    const getActiveSubs = () => {
+        const now = new Date().toISOString();
+        const activeSubs = (teamSubscriptions || []).filter(ts => TeamSubscription.isActive(ts, now));
+        return activeSubs;
+    }
+
+    const getAvailableSubTypes = () => {
+        const usedTypes = getActiveSubs().map(sub => getPlan(sub)?.type as string);
+        const types = getSubscriptionTypes().filter(t => !usedTypes.includes(t));
+        return types;
+    }
+
+    if (!showPaymentUI) {
+        return (<div></div>);
+    }
```

Comment:
Nit: This causes the page to be completely empty until all the data is fetched (noticeable delay).

<img width="1440" alt="Screenshot 2021-03-31 at 17 25 10" src="https://user-images.githubusercontent.com/599268/113170942-51f0c900-9247-11eb-8ae3-4ae489726519.png">

Instead, it might be nicer to always show at least the static content, e.g. the page heading & subtext (i.e. "All Teams" and the h2).

What do you think about moving this check inline in the main `return` content? (E.g. with something like `<headings/>{!!showPaymentUI && <dynamic-content/>}`.)

Reply:
Done!